### PR TITLE
SDK update and stop targeting .NET6 AB#7459622

### PR DESCRIPTION
**Pull request by omex-github-bot-internal**
update of NuGet packages

### CODE REVIEW

**Comment by K-Cully** [comment]
Earliest date for announced Net6 support drop was 21st March, so this shouldn't be merged yet.

**Comment by K-Cully** [comment]
Error: /usr/lib/mono/msbuild/Current/bin/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(241,5): error NETSDK1005: Assets file '/home/runner/work/Omex/Omex/obj/Microsoft.Omex.Extensions.Logging/project.assets.json' doesn't have a target for '.NETFramework,Version=v7.0'. Ensure that restore has run and that you have included 'net70' in the TargetFrameworks for your project. [/home/runner/work/Omex/Omex/src/Logging/Microsoft.Omex.Extensions.Logging.csproj]

**Comment by K-Cully** [comment]
Earlier error:
CSC : error CS9057: The analyzer assembly '/home/runner/work/Omex/Omex/bin/Microsoft.Omex.CodeGenerators.SettingsGen/Debug/netstandard2.0/Microsoft.Omex.CodeGenerators.SettingsGen.dll' references version '4.5.0.0' of the compiler, which is newer than the currently running version '4.4.0.0'. [/home/runner/work/Omex/Omex/src/CodeGenerators/SettingsGen.Example/Microsoft.Omex.SettingsGen.Example.csproj::TargetFramework=net7.0]

**Comment by K-Cully** [comment]
> Earlier error: CSC : error CS9057: The analyzer assembly '/home/runner/work/Omex/Omex/bin/Microsoft.Omex.CodeGenerators.SettingsGen/Debug/netstandard2.0/Microsoft.Omex.CodeGenerators.SettingsGen.dll' references version '4.5.0.0' of the compiler, which is newer than the currently running version '4.4.0.0'. [/home/runner/work/Omex/Omex/src/CodeGenerators/SettingsGen.Example/Microsoft.Omex.SettingsGen.Example.csproj::TargetFramework=net7.0]

There is a compiler conflict in the UT and example projects targeting Net Core with the new version of the Microsoft.CodeAnalysis projects when the SettingsGen project has these references updated. I haven't found the root cause yet so reverting those package updates for now, to unblock removal of NET6 support.

**Review by K-Cully — APPROVED** [review]

File: Directory.Build.props [hunk 1]
```diff
@@ -28,7 +27,7 @@
   </PropertyGroup>
   <PropertyGroup Label="Condition Variables" >
     <IsNetStandard>$(NetStandardVersions.Contains('$(TargetFramework)'))</IsNetStandard>
-    <IsNetCore>$(NetCoreVersions.Contains('$(TargetFramework)'))</IsNetCore>
+    <IsNetCore>$(DefaultNetCoreVersion.Contains('$(TargetFramework)'))</IsNetCore>
```
Comment (artempushkin) [inline Directory.Build.props:30]: I'm not sure this will work as expected, needs verification.

Comment (K-Cully) [reply Directory.Build.props]: re-added NetCoreVersions to align with previous behaviour

**Review by K-Cully — APPROVED** [review]

File: Directory.Build.props [hunk 2]
```diff
@@ -4,15 +4,14 @@
   </PropertyGroup>
 
   <PropertyGroup Label="Target Platforms" >
-    <DefaultNetCoreVersion>net7.0</DefaultNetCoreVersion>
-    <NetCoreVersions>$(DefaultNetCoreVersion);net6.0</NetCoreVersions>
+    <NetCoreVersions>net7.0</NetCoreVersions>
```
Comment (MaxDac) [inline Directory.Build.props:7]: Just to clarify, I'm considering this as working normally, having that with only one .NET Core version the default one should not be necessary.

**Review by MaxDac — APPROVED** [review]

**Pull request merged**